### PR TITLE
chore(datastore): add integration tests for datastore

### PR DESCRIPTION
**Pull request by Jordan-Nelson**
*Issue #, if available:*

*Description of changes:*
- add basic integration tests for the datastore save/query

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_datastore/example/integration_test/main_test.dart [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import 'save_test.dart' as save_tests;
+import 'query_test.dart' as query_tests;
+
+void main() async {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('amplify_datastore', () {
+    setUpAll(() async {
+      final authPlugin =
```
Comment (ragingsquirrel3) [inline packages/amplify_datastore/example/integration_test/main_test.dart:31]: nit: not an auth plugin?

File: packages/amplify_datastore/example/pubspec.yaml [hunk 2]
```diff
@@ -23,6 +23,10 @@ dependencies:
   #   path: ../../amplify_api
   amplify_flutter:
     path: ../../amplify_flutter
+  flutter_driver:
```
Comment (ragingsquirrel3) [inline packages/amplify_datastore/example/pubspec.yaml:26]: suggestion: I would define the min flutter version as 2.2.0 in env block above like https://github.com/aws-amplify/amplify-flutter/blob/main/packages/amplify_api/example/pubspec.yaml#L10 . The integration test driver is only in 2.0.0 + and the versions between 2 and 2.2 have some null safety issue that sometimes involves adding an override dep, so I think stating for flutter 2.2.0 + technically the correct thing to do. In theory, could make CI integration easier later by avoiding cryptic error messages.

File: packages/amplify_datastore/example/integration_test/query_test.dart [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    testWidgets('should return all data by default',
+        (WidgetTester tester) async {
+      Blog testBlog1 = Blog(name: 'blog one');
+      Blog testBlog2 = Blog(name: 'blog two');
+      Blog testBlog3 = Blog(name: 'blog three');
+      await Amplify.DataStore.save(testBlog1);
+      await Amplify.DataStore.save(testBlog2);
+      await Amplify.DataStore.save(testBlog3);
+      var blogs = await Amplify.DataStore.query(Blog.classType);
+      expect(blogs.length, 3);
```
Comment (HuiSF) [inline packages/amplify_datastore/example/integration_test/query_test.dart:40]: Should we also assert by comparing query result and the expected data?

Comment (Jordan-Nelson) [reply packages/amplify_datastore/example/integration_test/query_test.dart:40]: Updated 👍

File: packages/amplify_datastore/example/pubspec.yaml (hunk 2 above)
Comment (Jordan-Nelson) [reply packages/amplify_datastore/example/pubspec.yaml:27]: Updated 👍

File: packages/amplify_datastore/example/integration_test/main_test.dart [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import 'package:amplify_datastore_example/amplifyconfiguration.dart';
+
+import 'save_test.dart' as save_tests;
+import 'query_test.dart' as query_tests;
+
+void main() async {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('amplify_datastore', () {
+    setUpAll(() async {
```
Comment (ragingsquirrel3) [inline packages/amplify_datastore/example/integration_test/main_test.dart:30]: Would it be possible to add this block (inside an `if (!Amplify.isConfigured) `) for the query test and save test as well? That way, the test files can be run individually.

**Review by ragingsquirrel3 — APPROVED** [review]
Thanks for adding these! It passed locally for me on ios and android. I left one small comment, but otherwise looks great.

**Review by HuiSF — APPROVED** [review]
Thanks!

File: packages/amplify_datastore/example/integration_test/main_test.dart (hunk 4 above)
Comment (Jordan-Nelson) [reply packages/amplify_datastore/example/integration_test/main_test.dart:27]: Good call. Updated.

File: packages/amplify_datastore/example/integration_test/utils/setup_utils.dart [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import 'package:amplify_datastore/amplify_datastore.dart';
+import 'package:amplify_datastore_example/amplifyconfiguration.dart';
+import 'package:amplify_datastore_example/models/ModelProvider.dart';
+import 'package:amplify_flutter/amplify.dart';
+
+Future<void> configureAuth() async {
```
Comment (ragingsquirrel3) [inline packages/amplify_datastore/example/integration_test/utils/setup_utils.dart:6]: nit: not auth, could change to datastore or similar

Comment (Jordan-Nelson) [reply packages/amplify_datastore/example/integration_test/utils/setup_utils.dart]: Good catch. Copy/paste is not my friend today 😅

**Pull request merged**